Author: Kamil Swiatkowski <kamil.swiatkowski@gmail.com>
Description: Handle end of line comment on memcached.conf
Forwarded: not-needed
Origin: upstream, http://code.google.com/p/memcached/issues/detail?id=283

Index: memcached-1.4.20/scripts/start-memcached
===================================================================
--- memcached-1.4.20.orig/scripts/start-memcached
+++ memcached-1.4.20/scripts/start-memcached
@@ -62,7 +62,7 @@
     foreach my $line (<$etchandle>)
     {
         $line ||= "";
-        $line =~ s/\#.*//g;
+        $line =~ s/(?<!\\)\#[^\#]*//g;
         $line =~ s/\s+$//g;
         $line =~ s/^\s+//g;
         next unless $line;
