Description: show distribution on version
Updated: 2015-08-18

Index: memcached-1.4.25/memcached.c
===================================================================
--- memcached-1.4.25.orig/memcached.c	2016-01-22 09:44:15.965538856 -0500
+++ memcached-1.4.25/memcached.c	2016-01-22 09:44:15.961538810 -0500
@@ -2580,7 +2580,7 @@
     APPEND_STAT("pid", "%lu", (long)pid);
     APPEND_STAT("uptime", "%u", now - ITEM_UPDATE_INTERVAL);
     APPEND_STAT("time", "%ld", now + (long)process_started);
-    APPEND_STAT("version", "%s", VERSION);
+    append_stat("version", add_stats, c, "%s Ubuntu", VERSION);
     APPEND_STAT("libevent", "%s", event_get_version());
     APPEND_STAT("pointer_size", "%d", (int)(8 * sizeof(void *)));
 
@@ -3544,7 +3544,7 @@
 
     } else if (ntokens == 2 && (strcmp(tokens[COMMAND_TOKEN].value, "version") == 0)) {
 
-        out_string(c, "VERSION " VERSION);
+        out_string(c, "VERSION " VERSION " Ubuntu");
 
     } else if (ntokens == 2 && (strcmp(tokens[COMMAND_TOKEN].value, "quit") == 0)) {
 
